### PR TITLE
Adds QD and AS_QD emission from VariantAnnotator on GVS input

### PR DESCRIPTION
**Pull request by meganshand**
Previously, AS_QD was ignored by VariantAnnotator since the `annotate()` function always returned an empty map. Now AS_QD will be added.

Also QD could only use QUAL as part of the calculation, but GVS output only contains the `QUALapprox` annotation. This change allows QD to be calculated from `QUALapprox` if QUAL is not present.

### CODE REVIEW

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java [hunk 1]
```diff
@@ -364,6 +361,42 @@ public void testDeNovo() {
         Assert.assertEquals(lociWithHighConfidenceDeNovo, new int[] {10130767, 10197999});
     }
 
+    @Test
+    public void addQdToGvsVcf() {
+        final File inputVCF = getTestFile("gvsOutput.vcf");
```
Comment (jamesemery) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java:366]: Can you rename this to something that gives a little more info about where this is form/why its in the test suite?

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/allelespecific/AS_QualByDepth.java [hunk 2]
```diff
@@ -78,7 +76,9 @@ public List<VCFCompoundHeaderLine> getRawDescriptions() {
     public Map<String, Object> annotate(final ReferenceContext ref,
                                         final VariantContext vc,
                                         final AlleleLikelihoods<GATKRead, Allele> likelihoods ) {
-        return Collections.emptyMap();
+        // first vc is used for the annotation and the second vc here is used just to get the alleles, so in this case we can pass the same vc for both
+        Map<String, Object> annotation = finalizeRawData(vc, vc);
```
Comment (jamesemery) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/allelespecific/AS_QualByDepth.java:80]: wait so was this just totally broken except in combineGVCFs?

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java [hunk 3]
```diff
@@ -364,6 +361,42 @@ public void testDeNovo() {
         Assert.assertEquals(lociWithHighConfidenceDeNovo, new int[] {10130767, 10197999});
     }
 
+    @Test
+    public void addQdToGvsVcf() {
```
Comment (jamesemery) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java:365]: add a comment explaining that this has QUALapprox but no QUAL and that is what this is testing

**Review by jamesemery — COMMENTED** [review]
One comment about an exception that should/shouldn't be there and some minor comments about documentation but otherwise this looks reasonable to me. It looks like you are just bringing the QD annotation into parity with what it was already doing for AS_QD and it seems reasonable to me that it would operate this way

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/QualByDepth.java [hunk 4]
```diff
@@ -72,7 +73,14 @@ public Map<String, Object> annotate(final ReferenceContext ref,
...
         }
 
-        final double qual = -10.0 * vc.getLog10PError();
+        final double qual;
+        if (vc.hasLog10PError()) {
+            qual = -10.0 * vc.getLog10PError();
+        } else if (vc.hasAttribute(GATKVCFConstants.RAW_QUAL_APPROX_KEY)) {
+            qual = vc.getAttributeAsInt(GATKVCFConstants.RAW_QUAL_APPROX_KEY, 0);
+        } else {
+            throw new GATKException("No QUAL or QUAL_approx found in when calculating QD at " + vc.getContig() + ":" + vc.getStart());
```
[the remark this answers is not in the document]
Comment (meganshand) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/QualByDepth.java]: Yeah good point, not sure what I was trying to protect from here.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/QualByDepth.java [hunk 5]
```diff
@@ -72,7 +73,14 @@ public Map<String, Object> annotate(final ReferenceContext ref,
             return Collections.emptyMap();
         }
 
-        final double qual = -10.0 * vc.getLog10PError();
+        final double qual;
+        if (vc.hasLog10PError()) {
+            qual = -10.0 * vc.getLog10PError();
+        } else if (vc.hasAttribute(GATKVCFConstants.RAW_QUAL_APPROX_KEY)) {
+            qual = vc.getAttributeAsInt(GATKVCFConstants.RAW_QUAL_APPROX_KEY, 0);
```
[the remark this answers is not in the document]
Comment (meganshand) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/QualByDepth.java]: done

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/allelespecific/AS_QualByDepth.java (hunk 2 above)
Comment (meganshand) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/allelespecific/AS_QualByDepth.java:80]: I think so, but I'm worried that I'm missing something and this was left unimplemented intentionally for some reason. But the behavior from VariantAnnotator was undesirable in that if you asked for AS_QD it just wouldn't do anything.

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java (hunk 3 above)
Comment (meganshand) [reply src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java:366]: done

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java (hunk 1 above)
Comment (meganshand) [reply src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java]: done

**Review by jamesemery — APPROVED** [review]
Looks good to me from a code/gatk perspective. I'm not 1000% if there wasn't some good reason that these didn't work before but this is about as close to harmless of a change for these annotations as it gets. I think so you are probably good to merge.

**Pull request merged**